lib/ui/widgets/app_bar.dart

```diff
@@ -2,15 +2,94 @@ import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:reaxit/ui/theme.dart';
 
+abstract class AppbarAction {
+  Widget asIcon(BuildContext _);
+  Widget asMenuItem(BuildContext _);
+
+  const AppbarAction();
+}
+
+class IconAppbarAction extends AppbarAction {
+  final String text;
+  final String tooltip;
+  final IconData icon;
+  final void Function() onpressed;
+
+  const IconAppbarAction(this.text, this.icon, this.onpressed,
+      {String? tooltip})
+      : tooltip = tooltip ?? text;
+
+  @override
+  Widget asIcon(BuildContext _) {
+    return IconButton(
+      padding: const EdgeInsets.all(16),
+      onPressed: onpressed,
+      icon: Icon(icon),
+      tooltip: tooltip,
+    );
+  }
+
+  @override
+  Widget asMenuItem(BuildContext _) {
+    return IconButton(
+      onPressed: onpressed,
+      icon: Column(
+        children: [
+          Text(text),
+          Icon(icon),
+        ],
+      ),
```

Comment:
`IconButton` has separate `icon` and `text` fields that you can use that should fix the issue of overflowing buttons.

Reply:
My bad; `TextButton` has a `.icon` constructor which you can use to have a button with an icon and text.